woocommerce/woocommerce-gateway-stripe
# fixed bug that would require customer to click on use new payment met…

Fixes #2163 

## Changes proposed in this Pull Request:
Now the "Use new payment method" radio button only shows on pay order page when there are saved cards. Otherwise it is not needed

<img width="802" alt="Stripe pay for order page UPE - use a new payment method" src="https://user-images.githubusercontent.com/5509901/142291665-1683de4b-96ba-4549-9a2e-ea9974909f90.png">

## Testing instructions
- As a shopper
- With UPE active
- Remove any saved card in My account > Payment Methods
- Buy any product
- Go to checkout
- Select Stripe as the payment gateway
- Make sure the radio button "Use new payment method" does not show up
- Use a **valid** card `4242 4242 4242 4242`
- Click on save credit card
- Place order
- Must be successful
- Repeat but this time use the saved card to checkout
- The saved card radio button and the "Use new payment method" must show 
- Place order
- Must be successful

- Remove any saved card in My account > Payment Methods
- Buy any product
- Go to checkout
- Select Stripe as the payment gateway
- Make sure the radio button "Use new payment method" does not show up
- Use an **invalid** card `4000000000000002`
- Try to place order
- Go to My account > Orders
- Click on pay on the last order
- Select Stripe
- Make sure the radio button "Use new payment method" does not show up
- Select Stripe as the payment method
- Use a valid card `4242 4242 4242 4242`
- Place the order
- Must be successful
- Repeat but this time use the saved card to checkout on pay order page
- The saved card radio button and the "Use new payment method" must show 
- Place order
- Must be successful

---


**Post merge**
